Refactor:  separate UI from game logic

FlashGooden opened the pull request:
## Description
We were tasked with separating the game Logic from UI logic. I am submitting a draft to help us get on task.

## Spec

See Issue: [ISSUE_NUMBER](https://sparkbox.atlassian.net/browse/FSA2021-68)

## Validation

- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR has new code, so new tests were added or updated, and they pass.

### To Validate

1. Make sure all PR Checks have passed.
2. Pull down all related branches.
3. Navigate to `refactor--seperate-UI-from-game-logic`
3. run `npm start`
4. go to `localhost:3000`
5. interact with game UI the same way

FlashGooden committed:
Chore: Update server imports

-Previous version imports .js and .css files individually, this will import everything in the src folder instead.
-Changes imports in index.html to reflect above change.
-Updates Eslint config to operate in browser environment to allow manipulation of parts of the DOM

FlashGooden committed:
Refactor: Separate game logic and UI

-Previously all game and UI logic was mixed together in a single index.js file, this BEGINS to separate the two more with the creation of a game.js file and the removal of an explicit state object.
-In the previous code's place, creates a Game class that contains game logic and handleClickCell function that works with an instance of Game
-Also updates all tests to reflect these changes, including the import of the instance of the game into the test file

rise-erpelding reviewed (commented):
🤯 Is this the thing that was keeping us from being able to separate all of the js code into multiple files?

rise-erpelding commented on .eslintrc.js:
```diff
@@ -2,6 +2,7 @@ module.exports = {
   root: true,
   env: {
     node: true,
+    browser: true,
```
🤯 Is this the thing that was keeping us from being able to separate all of the js code into multiple files?

FlashGooden replied:
I think this just lets when we call 'document' in our index.js not have a linting issue. There was a separate linting rule which was messing up our imports. We need for right now to explicitly run the import with '.js' file at the end until we figure out how to mayyyyybe change the babel configuration?

rise-erpelding replied:
I didn't even catch the .js on the end, I'm glad you mentioned that! Ok, so maybe _that_ was the thing.

rise-erpelding commented on src/js/index.spec.js:
```diff
@@ -1,13 +1,9 @@
 import jsdom from 'jsdom';
-import handleClickCell from './Game/game';
+import { handleClickCell, game } from './Game/game';
```
> As far as I can tell, it only works if you import the actual variable, as the handleClickCell function uses the game variable directly from the game.js file.

Makes sense to me why you would do it this way.

FlashGooden replied:
Yea.... I think eventually we can import state. I'm just going with what works for right now.

mcksheridan committed:
Refactor: Update constructor function

Removed the argument that constructor function took and added currentPlayer and winner.
As the default value for currentPlayer is 'X,' I added that directly into the constructor
function and removed the new gamestate variable. new Game() will now load with the values
in gamestate by default. I updated everywhere gameState was referenced.

FlashGooden committed:
Feat: Add guard clause

-Changes to main that occurred before this branch was created included the addition of a guard clause.
-This incorporates these changes.
-Previous tests were also updated to correct an error, and minor refactoring was done in the test file to adhere better to the DRY principle.

rise-erpelding reviewed (approved):
Looks great! Tests pass, everything works, and we've incorporated the changes from main. Green light!
